client/src/js/references/components/__tests__/Create.test.js

```diff
@@ -0,0 +1,97 @@
+import { CreateReference, mapDispatchToProps } from "../Create";
+
+describe("<CreateReference />", () => {
+    const props = {
+        onSubmit: jest.fn()
+    };
+    let state;
+    const e = {
+        preventDefault: jest.fn(),
+        target: {
+            name: "name",
+            value: "foo",
+            error: "error"
+        }
+    };
+    beforeEach(() => {
+        state = {
+            name: "",
+            description: "",
+            dataType: "genome",
+            organism: "",
+            errorName: "",
+            errorDataType: ""
+        };
+    });
+
+    it("should render", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+    it("handleChange should return new state", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.find("ReferenceForm").simulate("change", e);
+        expect(wrapper.state()).toEqual({
+            ...state,
+            name: "foo"
+        });
+    });
+    it("handleChangeDataType should return dataType", () => {
+        const wrapper = shallow(<CreateReference {...props} />);
+        wrapper.find("DataTypeSelection").simulate("select", "foo");
+        expect(wrapper.state()).toEqual({
+            ...state,
+            dataType: "foo"
+        });
+    });
+    it("handleSubmit should return errorName when [this.state.name.length=0]", () => {
```

Comment:
It updates state and doesn't return anything. Revise your other test cases names too!

> handleSubmit() should update errorName in state when...